x/qgb/keeper/query_attestation.go

```diff
@@ -36,6 +36,9 @@ func (k Keeper) LatestAttestationNonce(
 	ctx context.Context,
 	request *types.QueryLatestAttestationNonceRequest,
 ) (*types.QueryLatestAttestationNonceResponse, error) {
+	if !k.CheckLatestAttestationNonce(sdk.UnwrapSDKContext(ctx)) {
```

Comment:
maybe unwrap once here?

Reply:
That one we need to unwrap because we need the sdk context, and we only have the `context.Context`, if I understand your question correctly

Reply:
ahh sorry, meant that we're unwrapping again on line 43, its 100% non-blocking

Reply:
Ah, we need to unwrap there too because we don't have the sdk context... We could put the context in a variable not to unwrap it twice if you prefer. Let me know if I need to make the change